rama-http/src/service/web/endpoint/response/datastar.rs

```diff
@@ -32,3 +33,66 @@ impl IntoResponse for DatastarScript {
         self.0.into_response()
     }
 }
+
+#[derive(Debug, Clone, Copy)]
+/// Datastar source map ready to be served.
+///
+/// Embedded version of the datastar source map file
+/// for improved debugging experience.
+///
+/// Learn more about datastar at <https://data-star.dev/>
+/// or in the book: <https://ramaproxy.org/book/web_servers.html#datastar>.
+pub struct DatastarSourceMap(Script<&'static str>);
+
+impl Default for DatastarSourceMap {
+    fn default() -> Self {
+        Self(Script(include_str!("./datastar.js.map")))
+    }
+}
+
+impl DatastarSourceMap {
+    #[inline]
+    /// Create a new [`DatastarSourceMap`] ready to be served.
+    #[must_use]
+    pub fn new() -> Self {
+        Default::default()
+    }
+}
+
+impl IntoResponse for DatastarSourceMap {
+    #[inline]
+    fn into_response(self) -> Response {
+        self.0.into_response()
+    }
+}
+
+/// Mount both datastar.js and datastar.js.map at the given prefix.
+///
+/// This is a convenience function that registers both the datastar script
+/// and its source map in one call.
+///
+/// # Example
+///
+/// ```rust,ignore
+/// use rama::http::service::web::{Router, response::datastar_assets};
+///
+/// let router = datastar_assets(Router::new(), "/assets");
+/// ```
+///
+/// This will serve:
+/// - `/assets/datastar.js` - The datastar script
+/// - `/assets/datastar.js.map` - The source map for debugging
+pub fn datastar_assets<State>(router: Router<State>, prefix: &str) -> Router<State>
```

Comment:
please remove this and also the use of it, it goes a bit against the principles we adhere to in rama abstractions. This adds to little value (if anything) to warrant a new level of abstraction. And revert the use from before.